### PR TITLE
Global average pooling

### PR DESCRIPTION
**Pull request by OverLordGoldDragon**
Building on #673, will detect `T == N` and take `mean()`, which is much faster than dot-multiply with `phi`, subsampling, `ifft`, and unpadding - while producing exact same result.

Didn't implement backend `mean`, proof of concept PR.

### CODE REVIEW

**Comment by OverLordGoldDragon** [comment]
> This seems too specific for a backend function. 

What do you mean?

> Why not simply add mean if at all necessary 

> give keepdimS (numpy convention) as an argument?

I can do that, but I thought kymatio's conservative on adding kwargs to backend not used anywhere (e.g. `fft` lacked `axis`)

**Comment by lostanlen** [comment]
LGTM. Did you observe a speedup?

**Comment by OverLordGoldDragon** [comment]
Never tried, surely `mean(U1)` beats `unpad(ifft(subsample(U1 * phi)))`

File: kymatio/scattering1d/frontend/numpy_frontend.py [hunk 1]
```diff
@@ -51,7 +51,7 @@ def scattering(self, x):
             size_scattering = 0
 
         S = scattering1d(x, self.backend.pad, self.backend.unpad, self.backend, self.J, self.psi1_f, self.psi2_f,
-                         self.phi_f, max_order=self.max_order, average=self.average, pad_left=self.pad_left,
```
Comment (MuawizChaudhary) [inline kymatio/scattering1d/frontend/numpy_frontend.py:54]: why is this only changed for the numpy frontend? Why not torch and tf?

**Review by MuawizChaudhary — COMMENTED** [review]
Can you test this actually gives speedup? How much?

**Comment by lostanlen** [comment]
This feature is a priority IMO but i think it needs to be paired with a benchmark and a tutorial demonstrating the use

**Pull request closed without merging**